### PR TITLE
sql: add `pg_lsn` data type

### PR DESCRIPTION
**Pull request by otan**
This commit adds the `pg_lsn` data type from postgres. It is an uint64
integer but displays/parses as `%X/%X`, where the high 32 bits and low
32 bits are separated by a `/`.

I believe using this data type is strictly superior to INT/STRING, as
there are custom operations that can be performed on it which I don't
think is appropriate for those types, hence the addition.

We're mostly using the automated tests to cover any poignant issues,
with some basic logic test operations added. Missing casts and operators
with pg_lsn will be added in a later PR.

Release note (sql change): Introduce the `pg_lsn` data type, which is
used to store the `lsn` associated with replication.

Epic: https://cockroachlabs.atlassian.net/browse/CRDB-26486

### CODE REVIEW

File: pkg/ccl/changefeedccl/avro_test.go [hunk 1]
```diff
@@ -479,21 +480,24 @@ func TestAvroSchema(t *testing.T) {
 		for _, typ := range append(types.Scalar, types.BoolArray, types.MakeCollatedString(types.String, `fr`), types.MakeBit(3)) {
 			switch typ.Family() {
 			case types.OidFamily:
-				continue
+				return
```
Comment (HonoreDB) [inline pkg/ccl/changefeedccl/avro_test.go:483]: This diff looks like it short-circuits the test as soon as we hit an OID type?

File: pkg/ccl/changefeedccl/avro_test.go [hunk 2]
```diff
@@ -463,6 +463,7 @@ func TestAvroSchema(t *testing.T) {
 			`INT8`:              `["null","long"]`,
 			`INTERVAL`:          `["null","string"]`,
 			`JSONB`:             `["null","string"]`,
+			`PG_LSN`:            `["null","string"]`,
```
Comment (HonoreDB) [inline pkg/ccl/changefeedccl/avro_test.go:466]: Would you mind also adding an example to value_goldens?

**Review by mgartner — APPROVED** [review]
:lgtm: 

> Reviewed 64 of 64 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/105031#-:-NYPBiH7AB-esGg1fCol:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @miretskiy, @otan, @rafiss, @renatolabs, @rharding6373, and @srosenberg)

___
*[`pkg/sql/randgen/datum.go` line 716 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/105031#-NYPBfcf7uEvedPKtUlE:-NYPBfcf7uEvedPKtUlF:bgc7vqv) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd0fd6e71ebb74eddb71ae30f8c884cd64fb0644/pkg/sql/randgen/datum.go#L716)):*
> ```Go
> 			tree.DMaxIPAddr,
> 		},
> 		types.PGLSNFamily: {
> ```

:heart: 
___
*[`pkg/sql/logictest/testdata/logic_test/pg_lsn` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/105031#-NYPAazvENBeNWpSzZkL:-NYPAazvENBeNWpSzZkM:b-17jcbk) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd0fd6e71ebb74eddb71ae30f8c884cd64fb0644/pkg/sql/logictest/testdata/logic_test/pg_lsn#L2)):*
> ```
> query T
> SELECT 'A01F0/1AAA'::pg_lsn
> ```

Can you test some invalid formats?
___
*[`pkg/sql/logictest/testdata/logic_test/pg_lsn` line 13 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/105031#-NYPAlB-7zlUXQtdCHuy:-NYPAlB-7zlUXQtdCHuz:b-kxgohf) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd0fd6e71ebb74eddb71ae30f8c884cd64fb0644/pkg/sql/logictest/testdata/logic_test/pg_lsn#L13)):*
> ```
> 
> query TT
> SELECT * FROM pg_lsn_table ORDER BY id
> ```

A lookup by PK might be a good test to have too.

File: pkg/ccl/changefeedccl/avro_test.go (hunk 1 above)
Comment (otan) [reply pkg/ccl/changefeedccl/avro_test.go:483]: ah a bad edit - thanks!

**Review by otan — COMMENTED** [review]
thanks for the quick reviews!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/105031#-:-NYPTDTiAEvr2B6vqhjE:b-1s3pdb)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @HonoreDB, @mgartner, @miretskiy, @rafiss, @renatolabs, @rharding6373, and @srosenberg)

___
*[`pkg/sql/logictest/testdata/logic_test/pg_lsn` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/105031#-NYPAazvENBeNWpSzZkL:-NYPTBtd9yzUM3Kfe3Ll:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd0fd6e71ebb74eddb71ae30f8c884cd64fb0644/pkg/sql/logictest/testdata/logic_test/pg_lsn#L2)):*
<details><summary><i>Previously, mgartner (Marcus Gartner) wrote…</i></summary><blockquote>

Can you test some invalid formats?
</blockquote></details>

Done.
___
*[`pkg/sql/logictest/testdata/logic_test/pg_lsn` line 13 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/105031#-NYPAlB-7zlUXQtdCHuy:-NYPSnk0AjhJcopn1LUy:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd0fd6e71ebb74eddb71ae30f8c884cd64fb0644/pkg/sql/logictest/testdata/logic_test/pg_lsn#L13)):*
<details><summary><i>Previously, mgartner (Marcus Gartner) wrote…</i></summary><blockquote>

A lookup by PK might be a good test to have too.
</blockquote></details>

Done.
___
*[`pkg/ccl/changefeedccl/avro_test.go` line 466 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/105031#gh-1235338784:-NYPTCXe8TNXFUPF0jnT:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd0fd6e71ebb74eddb71ae30f8c884cd64fb0644/pkg/ccl/changefeedccl/avro_test.go#L466)):*
<details><summary><i>Previously, HonoreDB (Aaron Zinger) wrote…</i></summary><blockquote>

Would you mind also adding an example to value_goldens?
</blockquote></details>

Done.

**Review by rharding6373 — APPROVED** [review]
:lgtm: Nice!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/105031#-:-NYQBtZG0k53meAu3B9C:b-8gk8uj)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @HonoreDB, @mgartner, @miretskiy, @otan, @rafiss, @renatolabs, and @srosenberg)

**Comment by otan** [comment]
i'm aware, was going to add it later (unless it's failing now): https://github.com/cockroachdb/cockroach/issues/105130

**Comment by rafiss** [comment]
It is failing now https://github.com/cockroachdb/cockroach/pull/105324#issuecomment-1601931310

**Pull request merged**